subprojects/dependency-management/src/main/java/org/gradle/internal/component/model/MultipleCandidateMatcher.java

```diff
@@ -217,6 +217,18 @@ private void disambiguateWithRequestedAttributeKeys() {
                 ImmutableAttributes candidateAttributeSet = candidateAttributeSets[c];
                 if (candidateAttributeSet.getAttributes().contains(extraAttribute)) {
                     any.set(c);
+                } else {
+                    // Attempt with attribute coercion
+                    AttributeValue<?> entry = candidateAttributeSet.findEntry(extraAttribute.getName());
```

Comment:
I wondered if only doing `findEntry` is not enough, including replacing the `contains` above.

After all, we are looking for extra attributes, whose value do not matter here. The only thing that made me not do it is the case where we find a value that we cannot coerce. Should we consider this a name clash and not the presence of the extra attribute.

Reply:
I think I would leave it like this. If the value cannot be coerced, something is fishy and we probably should not silently accept that.

Reply:
After discussion and because at that state, extra attributes have already been used in disambiguation, only the `findEntry` call is required with an `isPresent` check.